ViewControllers/DetailViewController.swift

```diff
@@ -147,14 +145,28 @@ class DetailViewController: UIViewController {
         super.viewDidLoad()
 
         guard let id = selectedMovieId else { return }
-        movieDataService.getMovieDetails(movieId: id) { [weak self] (movieDetails: MovieDetails?) in
-            if let movieDetails = movieDetails {
-                self?.movieDetails = movieDetails
+        
+        movieDataService.getMovieDetails(movieId: id) { [weak self] result in
+            guard let self = self else {return}
+            
+            switch result {
+            case .success(let details):
+                self.movieDetails = details
+            case .failure(let error):
+//                Here I wanted to display an alert controller with the info about the error. Not sure if error.localizedDescription will hold anything though.
```

Comment:
At the moment no since your error doesn't conform to `LocalizedError`. I've left a comment on how to do that :)